storage/src/persistent/mod.rs

```diff
@@ -125,20 +130,32 @@ pub struct PersistentStorage {
     merkle: Arc<RwLock<MerkleStorage>>,
 }
 
+pub enum StorageType{
+    Database,
+    Context,
+    ContxtAction,
+}
+
 impl PersistentStorage {
-    pub fn new(kv: Arc<DB>, clog: Arc<CommitLogs>) -> Self {
+    pub fn new(kv: Arc<DB>, kv_context: Arc<DB>, kv_context_actions: Arc<DB>, clog: Arc<CommitLogs>) -> Self {
         let seq = Arc::new(Sequences::new(kv.clone(), 1000));
         Self {
             clog,
             kv: kv.clone(),
+            kv_context: kv_context.clone(),
+            kv_context_actions: kv_context_actions.clone(),
             seq,
-            merkle: Arc::new(RwLock::new(MerkleStorage::new(kv))),
+            merkle: Arc::new(RwLock::new(MerkleStorage::new(kv_context))),
         }
     }
 
     #[inline]
-    pub fn kv(&self) -> Arc<DB> {
-        self.kv.clone()
+    pub fn kv(&self, storage: StorageType) -> Arc<DB> {
```

Comment:
@mat13mn 
wow, this PR looks easier then I thought :D,


I am just thinking about, if it would be possible to return here not `Arc<DB>` but something like:
```
Arc<dyn KeyValueStoreWithSchema<GenericSomething> + Sync + Send>
```

I mean `DB` is rocksdb struct, and `KeyValueStoreWithSchema` is our KV trait, so we could cover implementation detail, becasuse we use it like:
```
pub type BlockMetaStorageKV = dyn KeyValueStoreWithSchema<BlockMetaStorage> + Sync + Send;
...
pub struct BlockMetaStorage {
    kv: Arc<BlockMetaStorageKV>,
...
pub fn new(persistent_storage: &PersistentStorage) -> Self {
        BlockMetaStorage {
            kv: persistent_storage.kv(),
```


I am just thinking, never try it, and I dont know if it is even doable and not so generic-complicated

Reply:
i tried that but i run into problems, i ended up with 
```
    pub fn kv(&self, storage: StorageType) -> Arc<dyn KeyValueStoreWithSchema<KeyValueSchema<Key=[u8; 32],Value=Vec<u8>>> + Sync + Send> {
```

but compiler is still complaining 

```
storage/src/persistent/mod.rs:160:79
    |
160 |     pub fn kv(&self, storage: StorageType) -> Arc<dyn KeyValueStoreWithSchema<KeyValueSchema<Key=[u8; 32],Value=Vec<u8>>> + Sync + Send> {
    |                                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn KeyValueSchema<Key=[u8; 32],Value=Vec<u8>>`
    |
    = note: `#[warn(bare_trait_objects)]` on by default

error[E0038]: the trait `schema::KeyValueSchema` cannot be made into an object
   --> storage/src/persistent/mod.rs:160:47
    |
160 |     pub fn kv(&self, storage: StorageType) -> Arc<dyn KeyValueStoreWithSchema<KeyValueSchema<Key=[u8; 32],Value=Vec<u8>>> + Sync + Send> {
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `schema::KeyValueSchema` cannot be made into an object
    |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
   --> storage/src/persistent/schema.rs:14:8
    |
10  | pub trait KeyValueSchema {
    |           -------------- this trait cannot be made into an object...
...
14  |     fn descriptor(cache: &Cache) -> ColumnFamilyDescriptor {
    |        ^^^^^^^^^^ ...because associated function `descriptor` has no `self` parameter
...
18  |     fn name() -> &'static str;
    |        ^^^^ ...because associated function `name` has no `self` parameter

```

i`m not sure how to fix taht - cause i dont think changing this static function into non-static actually makes sense ...

Reply:
ok, ok, no problem, just leave it as it is now,
we will change it in future, when needed